aws/resource_aws_timestreamwrite_database.go

```diff
@@ -0,0 +1,148 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/timestreamwrite"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsTimestreamWriteDatabase() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsTimestreamWriteDatabaseCreate,
+		Read:   resourceAwsTimestreamWriteDatabaseRead,
+		Update: resourceAwsTimestreamWriteDatabaseUpdate,
+		Delete: resourceAwsTimestreamWriteDatabaseDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"database_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"kms_key_id": {
+				Type:     schema.TypeString,
+				Computed: true,
```

Comment:
if kms is not set does the api return some default key? if so add a check for that in the basic test. otherwise lets remove the computed attribute

Reply:
also if this is the value required is the Arn lets add `ValidateFunc: ValidateArn,`

Reply:
@DrFaust92 Thanks!

qq: The default is computed and set by AWS, using some defautl key the have, for example `arn:aws:kms:us-west-2:000560764000:key/4caec88d-1583-44f2-896f-208c625a8500`.
Which `TestCheckResourceAttr` should I write?
Just make sure its not null?

Reply:
see comment on Arn check, us the same logic for the `kms_id` field

Reply:
@DrFaust92 Ok, so with the arn, i can do:
`testAccCheckResourceAttrRegionalARN(resourceName, "arn", "timestream", fmt.Sprintf("database/%s", rName))`

Because I know what im expecting - `fmt.Sprintf("database/%s", rName)`

But I don't know what KMS id will come back, so I cant predict the ARN.

How should I proceed?

Sorry for being a noob at this 😬

Reply:
ok nvm!
will do `testAccMatchResourceAttrRegionalARN(resourceName, "kms_key_id", "kms", regexp.MustCompile(`key/.+`)),`